### DISCUSSION TITLE
Cannot import the library in a RN app using the "default" path

### DISCUSSION THREAD

**Discussion by mrAlmond** [callstack/react-native-builder-bob#644] [Q&A]
### Description

I've just created a test library using create-react-native-library, the example works correctly but if I import the library in an external RN project (an app) I cannot use the "default" import path:

`import { multiply } from 'testlib'`

but I have to add "src" ad the end to make it work:

`import { multiply } from 'testlib/src'`

This happens using the latest RN 0.75.2.
Is there a workaround for this?

Thank yuo.

### Packages

- [X] create-react-native-library
- [ ] react-native-builder-bob

### Selected options

npx create-react-native-library@latest testlib

### Link to repro

npx create-react-native-library@latest testlib

### Environment

System:
  OS: macOS 14.4.1
  CPU: (11) arm64 Apple M3 Pro
  Memory: 100.89 MB / 18.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 18.17.0
    path: ~/.local/share/rtx/installs/node/18.17.0/bin/node
  Yarn:
    version: 3.6.1
    path: ~/.local/share/rtx/installs/node/18.17.0/bin/yarn
  npm:
    version: 9.6.7
    path: ~/.local/share/rtx/installs/node/18.17.0/bin/npm
  Watchman: Not Found
Managers:
  CocoaPods:
    version: 1.15.2
    path: /opt/homebrew/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 23.5
      - iOS 17.5
      - macOS 14.5
      - tvOS 17.5
      - visionOS 1.2
      - watchOS 10.5
  Android SDK: Not Found
IDEs:
  Android Studio: 2023.1 AI-231.9392.1.2311.11255304
  Xcode:
    version: 15.4/15F31d
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: javac 17
    path: /Users/user/.local/share/rtx/installs/java/oracle-17/bin/javac
  Ruby:
    version: 3.2.2
    path: /Users/user/.local/share/rtx/installs/ruby/3.2.2/bin/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.3.1
    wanted: 18.3.1
  react-native:
    installed: 0.75.2
    wanted: 0.75.2
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: false
  newArchEnabled: false
iOS:
  hermesEnabled: Not found
  newArchEnabled: Not found

**Comment by satya164** [comment]
How are you installing the library in the app?

**Comment by mrAlmond** [comment]
In the package.json in the "dependencies" I've added this:

"testlib": "https://url/path/testlib.git",

The testlib library is retrieved from a private git repo.

**Comment by satya164** [comment]
> In the package.json in the "dependencies" I've added this:
> 
> 
> 
> "testlib": "https://url/path/testlib.git",
> 
> 
> 
> The testlib library is retrieved from a private git repo.

Depending on which package manager you use, you'd need to use either `prepare` or `prepack` scripts for the package to be built when installing from git. By default the project uses `prepare` - looks like it doesn't work with your package manager. Try changing it to `prepack` in your library's `package.json` and try.

More details here https://callstack.github.io/react-native-builder-bob/build#manual-configuration

**Comment by mrAlmond** [comment]
Thank you for the suggestion I wasn't aware of that option.
At the moment I'm using yarn 3.6.4.

In the library package.json I've set

```
"scripts": {
    "example": "yarn workspace testlib-example",
    "test": "jest",
    "typecheck": "tsc",
    "lint": "eslint \"**/*.{js,ts,tsx}\"",
    "clean": "del-cli android/build example/android/build example/android/app/build example/ios/build lib",
    "prepack": "bob build",
    "release": "release-it"
  },
```

Using prepack instead of prepare but unfortunately looks like the problem is still there. :-(

**Comment by satya164** [comment]
Can you create a demo repo with a test app and one with a test library that replicates enough your setup to reproduce the issue?

**Comment by mrAlmond** [comment]
For the test app and test library you can simply use create-react-native-library.
Unfortunately at the moment I can't setup a demo repo.

**Comment by atlj** [comment]
Hey, could you also fill which options you've picked? You've provided:
> Selected options
> npx create-react-native-library@latest testlib

but this is not enough for us to take care of this issue. We need more details so we can understand where this issue is coming from. Could you provide the project type, languages, and the example app?

**Comment by ridvanaltun** [comment]
Hey guys, I'm using yarn and I have very same issue. Reproducing the issue is very easy:

1. Create a new package with react-native-builder-bob. Settings are:
   `Native Module > Kotlin & Swift > Test App`
2. Upload the package to a Github repo.
3. Create a new React-Native app. `npx @react-native-community/cli@latest init AwesomeProject`
4. Install the package via Yarn. `yarn add package-name@git@github.com:username/package-name.git`
5. Try to use library.

Thats it. @atlj

**Comment by satya164** [comment]
According to Yarn docs, `prepack` script should run when installing from git (if you're using Yarn 3 or 4).

https://v3.yarnpkg.com/advanced/lifecycle-scripts
https://yarnpkg.com/advanced/lifecycle-scripts#prepack-and-postpack

Please confirm your project's Yarn version.

If it's not running `prepack`, then please open an issue in Yarn repo.

It's only an issue in Bob if `prepack` is running but it's not building the package.
